azext_iot/central/models/templatev1.py

```diff
@@ -0,0 +1,114 @@
+
+    def get_schema(self, name, is_component=False, identifier="") -> dict:
+        entities = self.components if is_component else self.interfaces
+        if identifier:
+            # identifier specified, do a pointed lookup
+            entry = entities.get(identifier, {})
+            return entry.get(name)
+
+        # find first matching name in any component
+        for entry in entities.values():
+            schema = entry.get(name)
+            if schema:
+                return schema
+
+        # not found
+        return None
+
+    def _extract_components(self, template: dict) -> dict:
+        try:
+            dcm = template.get("capabilityModel", {})
+            if dcm.get("contents"):
+                rootContents = dcm.get("contents", {})
+                components = [
+                    entity
+                    for entity in rootContents
+                    if entity.get("@type") == "Component"
+                ]
+
+                if components:
+                    return {
+                        component["name"]: self._extract_schemas(component)
+                        for component in components
+                    }
+                return {}
+            return {}
+        except Exception:
+            details = "Unable to extract schema for component from template '{}'.".format(
+                self.id
+            )
+            raise CLIError(details)
+
+    def _extract_root_interface_contents(self, dcm: dict):
+        rootContents = dcm.get("contents", {})
+        contents = [
+            entity for entity in rootContents if entity.get("@type") != "Component"
+        ]
+
+        return {"@id": dcm.get("@id", {}), "schema": {"contents": contents}}
+
+    def _extract_interfaces(self, template: dict) -> dict:
+        try:
+            interfaces = []
+            dcm = template.get("capabilityModel", {})
+
+            if dcm.get("contents"):
+                interfaces.append(self._extract_root_interface_contents(dcm))
+
+            if dcm.get("extends"):
+                interfaces.extend(dcm.get("extends"))
+
+            return {
+                interface["@id"]: self._extract_schemas(interface)
+                for interface in interfaces
+            }
+        except Exception:
+            details = "Unable to extract device schema from template '{}'.".format(
+                self.id
+            )
+            raise CLIError(details)
+
+    def _extract_schemas(self, entity: dict) -> dict:
+        if entity.get("schema"):
+            return {schema["name"]: schema for schema in entity["schema"]["contents"]}
+        else:
+            return {schema["name"]: schema for schema in entity["contents"]}
+
+    def _extract_schema_names(self, entity: dict) -> dict:
+        return {
+            entity_name: list(entity_schemas.keys())
+            for entity_name, entity_schemas in entity.items()
+        }
+
+    def _get_interface_list_property(self, property_name):
```

Comment:
Missing return type hint for consistency

Reply:
fixed